definitions/marts/looker_studio_marts/ecf/ls_ecf_2021_declarations_in_milestone_period.sqlx

```diff
@@ -0,0 +1,188 @@
+      Changed relevant profiles to use a concatenation of participant_profile_id & provider_id to remove declarations appearing against providers for participants with multiple in same milestone period but with a different lead provider.
+
+    1.3 2024-06-04 Tony Page
+      Made performance improvements and shortening of code by replacing 2 grouping CTEs with COUNT(*) over partitions and excluding unnecessary Participant-Provider combinations with a SUM() over partitions.
+      This also removed 2 dependencies.
+
+*/
+
+-- Gather milestone period and split into individual dates between start & end of period for use in joins later.
+WITH milestone_periods AS (
+  SELECT 
+     milestone_period_id
+    ,milestone_period_start_date
+    ,milestone_period_end_date
+    ,milestone_period_date
+  FROM 
+    static_tables.ecf_2021_milestone_periods
+  CROSS JOIN 
+    UNNEST(
+      GENERATE_DATE_ARRAY(milestone_period_start_date, milestone_period_end_date, INTERVAL 1 DAY)
+    ) AS milestone_period_date
+),
+
+-- Gather all ECF declarations where state is in 'paid', 'payable', 'eligible'
+ecf_declarations AS (
+  SELECT 
+     id AS declaration_id
+    ,user_id
+    ,participant_profile_id
+    ,course_identifier
+    ,cpd_lead_provider_id AS provider_id
+    ,cpd_lp_name AS provider_name
+    ,declaration_date
+    ,declaration_type
+    ,CASE
+      WHEN declaration_type = 'started' THEN 1
+      WHEN declaration_type = 'retained-1' THEN 2
+      WHEN declaration_type = 'retained-2' THEN 3
+      WHEN declaration_type = 'retained-3' THEN 4
+      WHEN declaration_type = 'retained-4' THEN 5
+      WHEN declaration_type = 'extended-1' THEN 6
+      WHEN declaration_type = 'extended-2' THEN 7
+      WHEN declaration_type = 'extended-3' THEN 8
+      WHEN declaration_type = 'completed' THEN 9
+    END AS declaration_type_sort_order
+    ,state AS declaration_state
+    ,evidence_held
+    ,cohort
+    ,statement_cohort
+    ,mentor_user_id
+    ,schedule_identifier
+  FROM
+    ${ref('ls_declarations_provider_names')}
+  WHERE
+    programme = 'ECF'
+    AND
+    state IN (
+       'paid'
+      ,'payable'
+      ,'eligible'
+    )
+    AND cohort = 2021
+),
+
+-- Combine data obtained so far and generate grouping columns
+  -- Milestone Period Group = user + course + lead provider + milestone period
+  -- Declaration Date Group = user + course + lead provider + declaration date
+combined_data AS (    
+  SELECT 
+    dec.*
+    ,CASE
+      WHEN course_identifier LIKE '%ecf-mentor%' THEN 'Mentor'
+      WHEN course_identifier LIKE '%ecf-induction%' THEN 'ECT'
+      ELSE ''
+    END AS participant_type
+    ,mp.milestone_period_id
+    ,mp.milestone_period_start_date
+    ,mp.milestone_period_end_date
+    -- Assess each grouping for quantity and apply an indicator flag (1 or 0) where not an allowed instance.
+    ,CASE 
+      WHEN COUNT(*) OVER (PARTITION BY user_id, course_identifier, provider_id, milestone_period_id) > 1 

```

Comment:
You're using user_id here but summing by participant_profile_id in the table below this one. Probably a good idea to be grouping consistently on the same type of id.